micro-benchmarks/DRB172-parallelfor-orig-no.c

```diff
@@ -0,0 +1,60 @@
+/**
+ * BT.C: This file is part of kernel of the NAS Parallel Benchmarks 3.0 bt suit.
+ * Tsan and Intel Inspector can not correctly analysis the nested parallel for, and report a false postive.
+*/
+
+#include <stdio.h>
+
+
+int main(int argh, char* argv[]){
+  int i,j,k,m;
+  int nthreads = 1;
+  double u[12][12][12][5];
+  double tmp1,tmp2,tmp3;
+  double fjac[12][12][12][5][5];
+
+  /*initial m */
+  #pragma omp for
+  for (i = 1; i < 12-1; i++) {
+    for (j = 1; j < 12-1; j++) {
+        for (k = 0; k < 12; k++){
+            for (m = 0; m < 5; m++){
+            u[i][j][k][m] = 2.5;
+      }
+    }
+    }
+  }
+    #pragma omp parallel
+    {
+  #pragma omp for
+  for (i = 1; i < 12-1; i++) {
+    for (j = 1; j < 12-1; j++) {
+      for (k = 0; k < 12; k++) {
+          tmp1 = 1.0 / u[i][j][k][0];
+          tmp2 = tmp1 * tmp1;
+          tmp3 = tmp1 * tmp2;
+          
+          fjac[i][j][k][0][0] = 0.0;
+          fjac[i][j][k][0][1] = 0.0;
+          fjac[i][j][k][0][2] = 0.0;
+          fjac[i][j][k][0][3] = 1.0;
+          fjac[i][j][k][0][4] = 0.0;
+
+          fjac[i][j][k][1][0] = - ( u[i][j][k][1]*u[i][j][k][3] ) * tmp2;
+          fjac[i][j][k][1][1] = u[i][j][k][3] * tmp1;
```

Comment:
@Yaying-llnl-summer, this can not be Data Race free. We are using `parallel for `without any synchronization construct. You can check for `if (i == 6 && j == 1 && k == 1) printf("%d,%d,%d; %ld\n",i,j,k,fjac[i][j][k][1][1]);` Alos, each iteration will not result into proper execution. 

Please have a look into this kernel.